Monal/Classes/MLIQProcessor.m

```diff
@@ -784,6 +785,20 @@ +(BOOL) processRosterWithAccount:(xmpp*) account andIqNode:(XMPPIQ*) iqNode
     }];
 $$
 
+$$class_handler(handleChangePassword, $$ID(xmpp*, account), $$ID(NSString*, newPass), $$ID(NSUUID*, uuid))
+    DDLogDebug(@"Handling password change");
+    [[MLXMPPManager sharedInstance] updatePassword:newPass forAccount:account.accountID];
+    MLPromise* promise = [account.promiseRegistry getPromise:uuid];
+    [promise resolve:nil];
+$$
+
+$$class_handler(handleChangePasswordInvalidation, $$ID(xmpp*, account), $$ID(NSUUID*, uuid))
+    DDLogDebug(@"Invalidating password change");
+    NSError* error = [NSError errorWithDomain:@"Monal" code:0 userInfo:@{NSLocalizedDescriptionKey: @"Could not change password"}];
```

Comment:
I haven't worked out how to expose the response IQ to the handler yet, but I plan to use it here when I work that out :)

Reply:
I think this could help:

> We should therefore create a category or subclass of NSError, somehow adding XMPPIQ into it. Maybe use multiple inheritance to create an XMPPErrorIQ class inheriting from both NSError and XMPPIQ. That would overwrite the error description method of NSError to use the HelperTools method extractXMPPError:withDescription: to get the error text.

Reply:
Ah well, now I understand!

You're confused what the invalidation handler is for.

For IQs the normal handler is called for `result` AND `error` IQs. The invalidation handler is only called, if the xmpp session is cleaned up or the IQ timed out (e.g. the result/error IQ will never be received).

I picked that design to allow other parts of our code to clean up in case an IQ result/error is missing.